pkg/cfn/builder/nodegroup.go

```diff
@@ -196,41 +184,134 @@ func (n *NodeGroupResourceSet) addResourcesForNodeGroup() error {
 			},
 		)
 	}
-	ngProps := map[string]interface{}{
-		"LaunchTemplate": map[string]interface{}{
-			"LaunchTemplateName": launchTemplateName,
-			"Version":            gfn.MakeFnGetAttString("NodeGroupLaunchTemplate.LatestVersionNumber"),
+
+	asg := nodeGroupResource(launchTemplateName, &vpcZoneIdentifier, tags, n.spec)
+	n.newResource("NodeGroup", asg)
+
+	return nil
+}
+
+// GetAllOutputs collects all outputs of the node group
+func (n *NodeGroupResourceSet) GetAllOutputs(stack cfn.Stack) error {
+	return n.rs.GetAllOutputs(stack)
+}
+
+func launchTemplateData(n *NodeGroupResourceSet) *gfn.AWSEC2LaunchTemplate_LaunchTemplateData {
```

Comment:
let's call it `newLaunchTemplateData`, as it constructs the data and returns a reference